implentio/ui-toolkit
# Feat: Create CollapsedMenu Component

## Description
Created `CollapsedMenu` component that consists of a caption and a sublist. Clicking on the trigger content will display the collapsed menu.
Props:

- triggerContent: the component passed in that will be the trigger content.
- caption: string
- subList: an array of objects, each object has two properties, text and onClick

## Link to JIRA ticket
[Link to JIRA ticket](https://implentio.atlassian.net/browse/IUT-29?atlOrigin=eyJpIjoiMjkyNTRlZWI0M2NkNGEzNmIxMTU1ZDU3ZGZhZjdkYWYiLCJwIjoiaiJ9)

## Applicable Screenshots
**Note: Icon component is used as an example trigger content to click on and show how collpased menu component renders**
<img width="244" alt="Screenshot 2024-02-09 at 10 46 01 AM" src="https://github.com/implentio/ui-toolkit/assets/104450960/4758b03c-c104-4377-8d88-db34be0418cc">
<img width="240" alt="Screenshot 2024-02-09 at 10 46 20 AM" src="https://github.com/implentio/ui-toolkit/assets/104450960/93bfca0c-8b3a-4e4b-9e69-d8e641d2fe74">
<img width="317" alt="Screenshot 2024-02-09 at 10 46 29 AM" src="https://github.com/implentio/ui-toolkit/assets/104450960/a91e24c8-03b4-4879-a461-fe40eb9f988c">
<img width="323" alt="Screenshot 2024-02-09 at 10 46 36 AM" src="https://github.com/implentio/ui-toolkit/assets/104450960/5259dc2d-1ea4-45a0-a6c3-ddb6ff4695fc">


## Testing Instructions
- Fetch and checkout my branch `git fetch && git checkout feat/IUT-29/CollapsedMenu`
- `npm run storybook`
- Check to see that when clicking on trigger content, the collapsed menu is displayed. Hover over sublist.

## Checklist:
Check what is applicable by putting an 'x' in between the []:
